graphql_compiler/compiler/ir_lowering_sql/constants.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2018-present Kensho Technologies, LLC.
+from collections import namedtuple
+from enum import Enum
+
+from sqlalchemy import String, func
+from sqlalchemy.sql import functions
+
+from ... import exceptions
+
+
+# These columns are reserved for the construction of recursive queries
+DEPTH_INTERNAL_NAME = u'__depth_internal_name'
+PATH_INTERNAL_NAME = u'__path_internal_name'
+LINK_INTERNAL_NAME = u'__link_internal_name'
+CYCLE_DETECTED_INTERNAL_NAME = u'__cycle_detected_internal_name'
```

Comment:
I'd prefer to construct a rule for which names are reserved compiler-wide, instead of banning *some specific* names, for *some specific* backends. We'll undoubtedly need more names, and we don't want migrating to a different backend to suddenly impose different naming constraints on the queries.

There will be some special fields (e.g. `__typename`) that start with double-underscores. How about reserving all names that start with triple-underscore, compiler-wide?

Reply:
That sounds a lot better, I can create a PR for that separately.

Reply:
Created with https://github.com/kensho-technologies/graphql-compiler/pull/130